### PR TITLE
[FLINK-34034][docs] Update the query hint docs to clarify the resolution of conflicts in kv hint and list hint

### PR DESCRIPTION
**Pull request by swuferhong**
## What is the purpose of the change

This pr is aims to update the query hint docs to clarify the resolution of conflicts in kv hint and list hint.

## Brief change log

- update the query hint docs to clarify the resolution of conflicts in kv hint and list hint.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector:no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? docs

### CODE REVIEW

File: docs/content/docs/dev/table/sql/queries/hints.md [hunk 1]
```diff
@@ -114,6 +114,42 @@ hintOption:
...
+
+```sql
+hintName '(' hintOption [, hintOption ]* ')'
+```
+
+With list hints, Flink resolves conflicts by adopting a first-accept strategy. This means that the 
+first specified hint in the list will take precedence and be effective.
+For example, consider the following SQL query with conflicting BROADCAST hints:
+```sql
+SELECT /*+ BROADCAST(t2, t1), BROADCAST(t1, t2) */ * FROM t1;
```
Comment (xuyangzhong) [inline docs/content/docs/dev/table/sql/queries/hints.md:147]: ditto.

File: docs/content.zh/docs/dev/table/sql/queries/hints.md [hunk 2]
```diff
@@ -108,6 +108,39 @@ hintOption:
...
+List 类型的查询提示使用如下语法：
+
+```sql
+hintName '(' hintOption [, hintOption ]* ')'
+```
+
+对于 List 类型的查询提示，Flink 会选择最先被采纳的查询提示。如下面具有相同 BROADCAST 提示的例子：
+
+```sql
+SELECT /*+ BROADCAST(t2, t1), BROADCAST(t1, t2) */ * FROM t1;
```
[the remark this answers is not in the document]
Comment (swuferhong) [reply docs/content.zh/docs/dev/table/sql/queries/hints.md:139]: done!

File: docs/content/docs/dev/table/sql/queries/hints.md (hunk 1 above)
Comment (swuferhong) [reply docs/content/docs/dev/table/sql/queries/hints.md:147]: done!

File: docs/content.zh/docs/dev/table/sql/queries/hints.md [hunk 3]
```diff
@@ -108,6 +108,39 @@ hintOption:
...
+### 查询提示使用中的冲突
+#### Key-value 类型查询提示的冲突处理
+Key-value 类型的查询提示使用如下语法：
+
+```sql
+hintName '(' optionKey '=' optionVal [, optionKey '=' optionVal ]* ')'
+```
+
+当 Key-value 类型的查询提示发生冲突时，Flink 会按顺序用后定义的查询提示覆盖前面定义的同名查询提示。
+对于下面的例子，包含同相同 key 的查询提示 'max-attempts'：
```
[the remark this answers is not in the document]
Comment (swuferhong) [reply docs/content.zh/docs/dev/table/sql/queries/hints.md:120]: done!

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by swuferhong** [comment]
@flinkbot run azure

**Review by LadyForest — COMMENTED** [review]
Hi @swuferhong thanks for your contribution! The fix looks good to me, and I just left one minor comment.

File: docs/content/docs/dev/table/sql/queries/hints.md [hunk 4]
```diff
@@ -114,6 +114,42 @@ hintOption:
     |   stringLiteral
 ```
 
+### Conflict Cases In Query Hints
```
[the remark this answers is not in the document]
Comment (swuferhong) [reply docs/content/docs/dev/table/sql/queries/hints.md:117]: After offline discussions with @LadyForest , we decided to keep this part of contents.

**Review by LadyForest — APPROVED** [review]
LGTM

**Pull request merged**